dvc/repo/collect.py

```diff
@@ -0,0 +1,55 @@
+import logging
+import os
+
+from dvc.path_info import PathInfo
+from dvc.tree.repo import RepoTree
+
+logger = logging.getLogger(__name__)
+
+
+def collect(
```

Comment:
Are you sure this couldn't be reorganized into meaningful parts? From the first glance looks cramped and hard to read.